Library/Homebrew/keg.rb

```diff
@@ -207,8 +207,8 @@ def ==(other)
   alias eql? ==
 
   def empty_installation?
-    Pathname.glob("#{path}/**/*") do |file|
-      next if file.directory?
+    Pathname.glob("#{path}/*") do |file|
+      return false if file.directory? && !file.children.reject(&:ds_store?).empty?
```

Comment:
Looks to me that it no longer recurses into subdirectories (`*` instead of `**/*`) and it's considered non-empty if it contains a directory that its not itself empty (while ignoring Mac special files named `.DS_Store`).